_adr/87/index.adoc

```diff
@@ -1,48 +1,442 @@
+A list of the metrics included in the remote-write configuration can be found in the https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk/blob/main/resources/prometheus/remote-write.yaml[Observability resources] repository.
+
+KAS Fleet Manager acts as a proxy to Observatorium.
+It calls the Observatorium _/query_ and _/query_range_ endpoints to retrieve metrics we consider as 'customer facing'.
+A list of the metrics retrieved by KAS Fleet Manager can be found in the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/client/observatorium/api.go[observatorium] client within the project. When new metrics are added to this list, they become available in the KAS Fleet Manager _/metrics/query_ and _/metrics/query_range_ endpoints. In order to make new metrics available to the _/metrics/federate_ endpoint, their metadata must be added to this https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/constants/metrics.go[file] in KAS Fleet Manager.
+
+KAS Fleet Manager controls the labels exposed for each metric returned in all endpoints. 
+Metrics retrieved from Observatorium are also filtered so that end users will only receive metrics for their own Kafka instance.
+Apart from these, no additional aggregation, filtering or modification of the metric is done by KAS Fleet Manager.
 
 ## Proposal
-// What is the decision being proposed
+### Metric and Label Naming
+All metric and label names must adhere to the policy stated in link:../../_ap/16/index.adoc[AP-16: Naming Policy for Customer Facing Metrics].
+
+### Kafka Metric Categories
+The customer facing Kafka metrics can be divided into multiple categories.
+Categorizing these metrics into logical groups can help us formulate the metric name and document it.
+
+|===
+|Category | Description | Sub-component (to be specified in the metric name)
+
+|Cluster
+|General Kafka metrics not specific to any other sub-components.
+|None
+
+|Broker
+|Metrics associated to Kafka brokers
+|broker
+
+|Topic
+|Metrics associated to Kafka topics
+|None (the topic metrics already have a 'topic' label so we should avoid using this in the metric name, as per the naming policy)
+|===
+
+Examples of sub-component usage in the metric name:
+
+* For *Cluster* metrics, the metric prefix would be `rhosak_`.
+* For *Broker* metrics, the metric prefix would be `rhosak_broker_`.
+* For *Topic* metrics, the metric prefix would be `rhosak_`.
+
+### Converted Metrics
+At minimum, each metric should have an `instance_name` label that identifies the Kafka instance.
+This would be useful if a user has multiple Kafka instances.
+If they want to integrate all of their Kafka metrics into one monitoring system, the `instance_name` can act as an identifier.
+
+#### Cluster Metrics
+|===
+|Old Metric Name | Revised Metric Name | Type | Old Labels | Revised Labels | Description
+
+|kafka_instance_max_message_size_limit
+|rhosak_message_size_limit_bytes
```

Comment:
isn't `rhosak` product specific? I am thinking of upstream / product and someone in the external might get confused if they see a metric with that name.

Reply:
I think this is specific to our product though. Unless there is a plan to make the [observability-resources-mk](https://github.com/bf2fc6cc711aee1a0c2a/observability-resources-mk) configuration public :thinking: I don't remember any discussion around the plan/alternative for that yet?

Reply:
There won't be any rename happening on the way KFM retrieve the metrics?

Reply:
> no there shouldn't be, KFM should only act as a proxy. It shouldn't rename any metric/its labels.

+1. This is correct.

What I was saying is that KFM will then have to start requesting for the metrics using the revised name. 
Instead of `kafka_controller_kafkacontroller_global_partition_count`, it'll use `rhosak_broker_global_partition_total`.

Reply:
ah good point! we would also need to change it in the metrics metadata: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/constants/metrics.go

We could make the prefix configurable if we want to have separate names between upstream/downstream. The upstream/downstream observability-resources-mk config would have different names for the user facing metrics. Do we have an upstream name yet? :thinking: 

Otherwise, we can have a generic name for both that's not product specific. Do you have anything in mind? :slightly_smiling_face:

Reply:
> We could make the prefix configurable if we want to have separate names between upstream/downstream.

Yes. 

> We could make the prefix configurable if we want to have separate names between upstream/downstream. The upstream/downstream observability-resources-mk config would have different names for the user facing metrics. 

+1.  

> Do we have an upstream name yet?

As far as I remember `kclerk` was the most voted one but I've seen mention of `bf2` (e.g org.bf2, bf2.dev etc) in some other projects. @emmanuelbernard might help us here.

> Otherwise, we can have a generic name for both that's not product specific. Do you have anything in mind?

Other than the `kafka` prefix, I do not have anything in mind.

Maybe @tombentley and @emmanuelbernard have a better ideas or using `rhosak` prefix is just fine overall.

Reply:
Would there be a preference to use `rhosak_` for downstream or would `kas_` be ok to use in general?

Reply:
as per discussion in the architecture call, we will be using `kas_` for both upstream and downstream.

Reply:
+1 on that as well and thanks for the update @JameelB and for following up on this one.

Reply:
All references of the revised metric names has now been updated to use `kas_` prefix.